packages/cli-hydrogen/src/cli/services/deploy/config.integration.test.ts

```diff
@@ -1,44 +1,46 @@
-      await expect(getDeployConfig({...defaultConfig, path: tmpDir})).rejects.toThrow(git.OutsideGitDirectoryError())
-    })
-  })
-  it('throws if no commit', async () => {
-    await file.inTemporaryDirectory(async (tmpDir) => {
-      await git.initializeRepository(tmpDir)
-      await expect(getDeployConfig({...defaultConfig, path: tmpDir})).rejects.toThrow(git.NoCommitError())
+  describe("User doesn't want to initialize new repository", () => {
+    it('silent abort since outside git directory', async () => {
+      await file.inTemporaryDirectory(async (tmpDir) => {
+        mockedGitInitPrompt.mockResolvedValue(false)
+
+        await expect(getDeployConfig({...defaultConfig, path: tmpDir})).rejects.toThrow(new error.AbortSilent())
+      })
     })
   })
-  it('extract basic information from git', async () => {
-    const authorName = 'integration test'
-    const authorEmail = 'integration@test.com'
-    const commitMessage = 'integration test message'
 
-    await file.inTemporaryDirectory(async (tmpDir) => {
-      await git.initializeRepository(tmpDir)
-      await file.touch(`${tmpDir}/integration.txt`)
-      const commitSha = await git.commitAll(commitMessage, {
-        directory: tmpDir,
-        author: `${authorName} <${authorEmail}>`,
-      })
-      const config = await getDeployConfig({...defaultConfig, path: tmpDir})
+  describe('User does initialize new repository', () => {
+    it('extract basic information from git', async () => {
+      await file.inTemporaryDirectory(async (tmpDir) => {
+        mockedGitInitPrompt.mockResolvedValue(true)
+        const defaultBranchConfig = await git.factory().getConfig('init.defaultBranch')
+        // Adding something to commit
+        await file.touch(`${tmpDir}/integration.txt`)
 
-      expect(config.commitAuthor).toBe(authorName)
-      expect(config.commitMessage).toBe(commitMessage)
-      expect(config.commitRef).toBe('refs/heads/master')
-      expect(config.commitSha).toBe(commitSha)
-      expect(config.deploymentToken).toBe(defaultConfig.deploymentToken)
-      expect(config.dmsAddress).toBe(defaultConfig.dmsAddress)
-      expect(config.healthCheck).toBe(defaultConfig.healthCheck)
-      expect(config.path).toBe(tmpDir)
-      expect(new Date(config.timestamp).getTime()).toBeLessThan(new Date().getTime())
+        const config = await getDeployConfig({...defaultConfig, path: tmpDir})
+
+        expect(config.commitMessage).toBe('Initial Commit')
+        expect(config.commitRef).toBe(`refs/heads/${defaultBranchConfig.value ?? 'master'}`)
+        expect(config.deploymentToken).toBe(defaultConfig.deploymentToken)
+        expect(config.dmsAddress).toBe(defaultConfig.dmsAddress)
+        expect(config.healthCheck).toBe(defaultConfig.healthCheck)
+        expect(config.path).toBe(tmpDir)
+        expect(new Date(config.timestamp).getTime()).toBeLessThan(new Date().getTime())
+        // can't test config.commitAuthor & config.commitSha
```

Comment:
Since this integration test can be run anywhere, there's certain things we can't test for due to flakiness. The commit author could or couldn't be set and there is no clear fallback like for default branch.

Reply:
👍🏻 adding the context of why we can't test those things to the comment will be helpful